public/JS Files/restaurant.js

```diff
@@ -0,0 +1,42 @@
+
+
+let url = window.location.search
+let urlArray = url.split('?')
+let id = urlArray.pop()
+
+let htmlName = document.getElementById('name')
+let htmlNotes = document.getElementById('notes')
+let htmlPhone = document.getElementById('phone')
+let htmlAddress = document.getElementById('address')
+let htmlHours = document.getElementById('hours')
+
+async function restDetails() {
+    let restSpecs = await fetch(`/api/${id}.json`)
+        .then(res => res.json())
+        .then((jsonObj) => {
+            return jsonObj
+        })
+
+    htmlName.textContent = restSpecs.name
+    htmlNotes.innerHTML = restSpecs.notes.join('<br/><br/>')
+    htmlPhone.textContent = restSpecs.phone
+    htmlAddress.textContent = restSpecs.address
+    htmlHours.textContent = restSpecs.hours
+
+    fetch(`https://nominatim.openstreetmap.org/search/?q=${restaurantDetails.address}&format=json`).then((data) => {
```

Comment:
A typo here, ```restaurantDetails``` should be ```restSpecs```, this and the change below will fix the issue causing the map to not render or place a marker